aten/src/ATen/native/mkldnn/MKLDNNConversions.cpp

```diff
@@ -15,7 +15,12 @@ Tensor mkldnn_to_dense(const Tensor& mkldnn_tensor) {
   Tensor cpu_tensor = at::empty(
     std::vector<int64_t>(dims.begin(), dims.end()),
     mkldnn_tensor.options().layout(c10::kStrided));
-  stensor.to_public(cpu_tensor.template data_ptr<float>());
+  if (stensor.is_empty()) return cpu_tensor;
+  auto pub_tensor =
+      cpu_tensor.scalar_type() == ScalarType::Float
+          ? stensor.to_public(cpu_tensor.template data_ptr<float>())
+          : stensor.to_public(cpu_tensor.template data_ptr<BFloat16>());
```

Comment:
Shall we add an assertion check to guarantee that scalar_type could be either `ScalarType::Float` or `ScalarType::BFloat16", no others? Then you may consider to simplify the code with `ScalarTypeToCPPType`, without the need of a ternary operator.

Reply:
Oops, this bf16 stuff should not be in the public pr